### DISCUSSION TITLE
How to change Progress bar color?

### DISCUSSION THREAD

**Discussion by Hisui02** [shadcn-ui/ui#1454] [Q&A] [answered]
Hello, I am trying to build two mirrored progress bars, showing two teams current gold, I would like to change every bar to the team's color but as far as I know there is no prop that I can use to do that, I have opened the component file and the bar background has a bg-primary written and does not seem to be configurable. If anyone can tell me how to change it or even how to customize the component to add it I would be very grateful. Thanks

Note: Currently I have cloned and exported the component inside the file, changing the name to ProgressBlue and ProgressRed and updating the Tailwind background class, is there a way to do it dinamically?

**Comment by NikharPandya** [comment]
@Hisui02 have you tried using accent property from tailwindcss ? https://tailwindcss.com/docs/accent-color 
you could add it to the progress root component https://github.com/shadcn-ui/ui/blob/main/apps/www/registry/default/ui/progress.tsx

**Comment by Hisui02** [comment]
Tailwind CSS accent property does not work and adding it to the component was what I was trying but honestly I got lost with the typescript and couldn't add the prop properly

**Comment by itaim18** [comment, marked as answer]
I would do something like 
```tsx
"use client";

import * as React from "react";
import * as ProgressPrimitive from "@radix-ui/react-progress";

import { cn } from "@/lib/utils";

interface CustomProgressProps
  extends React.ComponentPropsWithoutRef<typeof ProgressPrimitive.Root> {
  indicatorColor: string;
}
const Progress = React.forwardRef<
  React.ElementRef<typeof ProgressPrimitive.Root>,
  CustomProgressProps
>(({ className, value, indicatorColor, ...props }, ref) => (
  <ProgressPrimitive.Root
    ref={ref}
    className={cn(
      "relative h-4 w-full overflow-hidden rounded-full bg-secondary",
      className
    )}
    {...props}
  >
    <ProgressPrimitive.Indicator
      className={`h-full w-full flex-1  transition-all ${indicatorColor}`}
      style={{ transform: `translateX(-${100 - (value || 0)}%)` }}
    />
  </ProgressPrimitive.Root>
));
Progress.displayName = ProgressPrimitive.Root.displayName;

export { Progress };

```
and would treat this in my code as such: 
```tsx
<Progress
            value={50}
            className="w-[60%]"
            indicatorColor="bg-blue-300" />
```
If the root's component isn't changing as much I would suggest indicating the styles to the indicator and maintain a static style to the root
Hope this helps :)

**Reply by Hisui02** [reply]
Thank you so much, I was having so much problem with the TS

**Reply by AdityaSetyadi** [reply]
I've tested it thoroughly, and everything is working perfectly. However, I modified the `interface` declaration by replacing it with `type`.

```tsx
"use client";

import * as React from "react";
import * as ProgressPrimitive from "@radix-ui/react-progress";

import { cn } from "@/lib/utils";

type CustomProgressProps = React.ComponentPropsWithoutRef<
  typeof ProgressPrimitive.Root
> & {
  indicatorColor: string;
};

const Progress = React.forwardRef<
  React.ElementRef<typeof ProgressPrimitive.Root>,
  CustomProgressProps
>(({ className, value, indicatorColor, ...props }, ref) => (
  <ProgressPrimitive.Root
    ref={ref}
    className={cn(
      "relative h-4 w-full overflow-hidden rounded-full bg-secondary",
      className
    )}
    {...props}
  >
    <ProgressPrimitive.Indicator
      className={`h-full w-full flex-1 transition-all ${indicatorColor}`}
      style={{ transform: `translateX(-${100 - (value || 0)}%)` }}
    />
  </ProgressPrimitive.Root>
));
Progress.displayName = ProgressPrimitive.Root.displayName;

export { Progress };

```

**Comment by hkaya15** [comment]
I think "indicatorColor" doesn't work anymore. It use primary color from global.css. Is there any suggestion?

**Reply by hkaya15** [reply]
`<ProgressPrimitive.Root
		ref={ref}
		className={cn(
			'relative h-2 w-full overflow-hidden rounded-full bg-secondary',
			className
		)}
		{...props}>
		<ProgressPrimitive.Indicator
			className='h-full w-full flex-1 bg-indigo-500 transition-all'
			style={{ transform: `translateX(-${100 - (value || 0)}%)` }}
		/>
	</ProgressPrimitive.Root>`

**Reply by hkaya15** [reply]
I use like that and send props for bg-primary for now

**Comment by roninCode** [comment]
If you're just trying to change the color of the indicator using tailwind, you could use tailwind to target the child of Progress (the indicator).

[Per option #2](https://stackoverflow.com/a/71795600)
you can use `[&>*]:` to target the children of a component. 

In this case you could do something like:

```
  <Progress  className='[&>*]:bg-red-600' value={50} max={100} />
```
and it would change the indicator to red.

More of a quick and dirty way to do it if you don't want to create a whole new wrapper component

**Reply by thallysondias** [reply]
Clean solution!

**Reply by castroalves** [reply]
Best answer! Thanks for that!

**Reply by PsSave** [reply]
obrigado!

**Reply by Mohraf** [reply]
I love this solution. Worked well for me

**Comment by trong-codelink** [comment]
Another simple way is to add a className directly to `ProgressPrimitive.Indicator`

I add new prop named `indicatorClassName` to parent component and pass it down to `ProgressPrimitive.Indicator`

```typescript
"use client"

import * as React from "react"
import * as ProgressPrimitive from "@radix-ui/react-progress"

import { cn } from "@/lib/utils"

const Progress = React.forwardRef<
  React.ElementRef<typeof ProgressPrimitive.Root>,
  React.ComponentPropsWithoutRef<typeof ProgressPrimitive.Root> & {
    indicatorClassName?: string
  }
>(({ className, value, indicatorClassName, ...props }, ref) => (
  <ProgressPrimitive.Root
    ref={ref}
    className={cn(
      "relative h-4 w-full overflow-hidden rounded-full bg-secondary",
      className
    )}
    {...props}
  >
    <ProgressPrimitive.Indicator
      className={cn(
        "h-full w-full flex-1 bg-primary transition-all",
        indicatorClassName
      )}
      style={{ transform: `translateX(-${100 - (value || 0)}%)` }}
    />
  </ProgressPrimitive.Root>
))
Progress.displayName = ProgressPrimitive.Root.displayName

export { Progress }
```

**Comment by flan02** [comment]
I'd recommend you all to modify the <Progress /> component from shadcn/ui something like that
```tsx
"use client"

import * as React from "react"
import * as ProgressPrimitive from "@radix-ui/react-progress"

import { cn } from "@/lib/utils"

const Progress = React.forwardRef<
  React.ElementRef<typeof ProgressPrimitive.Root>,
  React.ComponentPropsWithoutRef<typeof ProgressPrimitive.Root>
>(({ className, value, ...props }, ref) => (
  <ProgressPrimitive.Root
    ref={ref}
    className={cn(
      "relative h-2 w-full overflow-hidden rounded-full bg-primary/20",
      className
    )}
    {...props}
  >
    <ProgressPrimitive.Indicator
      className="h-full w-full flex-1 bg-orange-400 transition-all" // * MODIFY YOUR COLOR HERE
      style={{ transform: `translateX(-${100 - (value || 0)}%)` }}
    />
  </ProgressPrimitive.Root>
))
Progress.displayName = ProgressPrimitive.Root.displayName

export { Progress }
```

**Comment by Edmond-XavierCollot** [comment]
Update `bg-primary` to `bg-current` on the Indicator to allow color inheritance:

```tsx
<ProgressPrimitive.Indicator
  data-slot="progress-indicator"
  className="bg-current h-full w-full flex-1 transition-all"
  style={{ transform: `translateX(-${100 - (value || 0)}%)` }}
/>
```

And set the color with `text`:

```tsx
<Progress value={50} className="text-blue-500" />
```
